src/runner/settings_window.cpp

```diff
@@ -4,89 +4,86 @@
+  result.SetNamedValue(L"powertoys", get_power_toys_settings());
   return result;
 }
 
-void dispatch_json_action_to_module(const json::value& powertoys_configs) {
-  for (auto powertoy_element : powertoys_configs.as_object()) {
-    std::wstringstream ws;
-    ws << powertoy_element.second;
-    if (modules().find(powertoy_element.first) != modules().end()) {
-      modules().at(powertoy_element.first).call_custom_action(ws.str());
+void dispatch_json_action_to_module(const json::JsonObject& powertoys_configs) {
+  for (const auto& powertoy_element : powertoys_configs) {
+    const std::wstring name{powertoy_element.Key().c_str()};
+    if (modules().find(name) != modules().end()) {
+      const auto element = powertoy_element.Value().Stringify();
+      modules().at(name)->call_custom_action(element.c_str());
     }
   }
 }
 
 void send_json_config_to_module(const std::wstring& module_key, const std::wstring& settings) {
   if (modules().find(module_key) != modules().end()) {
-    modules().at(module_key).set_config(settings);
+    modules().at(module_key)->set_config(settings.c_str());
   }
 }
 
-void dispatch_json_config_to_modules(const json::value& powertoys_configs) {
-  for (auto powertoy_element : powertoys_configs.as_object()) {
-    std::wstringstream ws;
-    ws << powertoy_element.second;
-    send_json_config_to_module(powertoy_element.first, ws.str());
+void dispatch_json_config_to_modules(const json::JsonObject& powertoys_configs) {
+  for (const auto& powertoy_element : powertoys_configs) {
+    const auto element = powertoy_element.Value().Stringify();
+    send_json_config_to_module(powertoy_element.Key().c_str(), element.c_str());
   }
 };
 
 void dispatch_received_json(const std::wstring &json_to_parse) {
-  json::value j = json::value::parse(json_to_parse);
-  for(auto base_element : j.as_object()) {
-    if (base_element.first == L"general") {
-      apply_general_settings(base_element.second);
-      std::wstringstream ws;
-      ws << get_all_settings();
-      if (current_settings_ipc != NULL) {
-        current_settings_ipc->send(ws.str());
-      }
-    } else if (base_element.first == L"powertoys") {
-      dispatch_json_config_to_modules(base_element.second);
-      std::wstringstream ws;
-      ws << get_all_settings();
-      if (current_settings_ipc != NULL) {
-        current_settings_ipc->send(ws.str());
-      }
-    } else if (base_element.first == L"refresh") {
-      std::wstringstream ws;
-      ws << get_all_settings();
-      if (current_settings_ipc != NULL) {
-        current_settings_ipc->send(ws.str());
-      }
-    } else if (base_element.first == L"action") {
-      dispatch_json_action_to_module(base_element.second);
+  const json::JsonObject j = json::JsonObject::Parse(json_to_parse);
+  for(const auto & base_element : j) {
+    const auto name = base_element.Key();
+    const auto value = base_element.Value();
+
+    if (name == L"action") {
```

Comment:
can you keep the original if/else structure here?

Reply:
@bzoz why?

Reply:
The old one is a plain "if/else if " list, this one has a continue inside one of the if-s. I find the old one easier to follow.